py/server/deephaven/time.py

```diff
@@ -155,1884 +160,595 @@ def time_zone_alias_rm(alias: str) -> bool:
 
 # endregion
 
-# region Conversions: Time Units
-
 
-def micros_to_nanos(micros: int) -> int:
-    """ Converts microseconds to nanoseconds.
+# region Conversions: Python To Java
 
-    Args:
-        micros (int): Microseconds to convert.
-
-    Returns:
-        NULL_LONG if the input is NULL_LONG; otherwise the input microseconds converted to nanoseconds.
 
-    Raises:
-        DHError
+def to_j_time_zone(tz: Union[None, TimeZone, str, datetime.tzinfo, datetime.datetime, pandas.Timestamp]) -> \
+        Optional[TimeZone]:
     """
-    try:
-        return _JDateTimeUtils.microsToNanos(micros)
-    except Exception as e:
-        raise DHError(e) from e
-
-
-def millis_to_nanos(millis: int) -> int:
-    """ Converts milliseconds to nanoseconds.
+    Converts a time zone value to a Java TimeZone.
+    Time zone values can be None, a Java TimeZone, a string, a datetime.tzinfo, a datetime.datetime,
+    or a pandas.Timestamp.
 
     Args:
-        millis (int): Milliseconds to convert.
+        tz (Union[None, TimeZone, str, datetime.tzinfo, datetime.datetime, pandas.Timestamp]): A time zone value.
+            If None is provided, None is returned.
+            If a string is provided, it is parsed as a time zone name.
 
     Returns:
-        NULL_LONG if the input is NULL_LONG; otherwise the input milliseconds converted to nanoseconds.
+        TimeZone
 
     Raises:
         DHError
     """
     try:
-        return _JDateTimeUtils.millisToNanos(millis)
+        if tz is None or pandas.isnull(tz):
+            return None
+        elif isinstance(tz, TimeZone.j_type):
+            return tz
+        elif isinstance(tz, str):
+            return _JDateTimeUtils.parseTimeZone(tz)
+        elif isinstance(tz, datetime.tzinfo):
+            return _JDateTimeUtils.parseTimeZone(str(tz))
+        elif isinstance(tz, datetime.datetime):
+            if not tz.tzname():
+                return _JDateTimeUtils.parseTimeZone(tz.astimezone().tzname())
+
+            return _JDateTimeUtils.parseTimeZone(tz.tzname())
+        else:
+            raise Exception("Unsupported conversion: " + str(type(tz)) + " -> TimeZone")
```

Comment:
Should we use DHError directly to avoid one level of wrapping for the unsupported conversions in these methods?

Reply:
I'd prefer the built-in `TypeError` which is more precise although I could live with `DHError`.